### ISSUE TITLE
I can't get anything by using this demo(new and old demo) in Android 6.0? why

### ISSUE THREAD

**Issue by pangpangshui** [188080501/JQQRCode#3]
I get some debug info, but it is so long that I just clip some like this:

> _W GLConsumer: [SurfaceTexture-0-21322-0] bindTextureImage: clearing GL error: 0x502_

Can you give me a hand? And may  I ask which system version in your Android phone?

**Comment by 188080501** [comment]
What's your android device model?

**Comment by pangpangshui** [comment]
Redmi 2A with Android 5.1 and Redmi Note 4 with Android 6.0 
@188080501

**Comment by pangpangshui** [comment]
Also I found the reason why this bug happend may be the Android 6.0 system. Just like this
http://www.bytran.org/androidmpermissions.htm

**Comment by 188080501** [comment]
When you run demo, is a black screen, or camera image displayed, but can not analyse the data?

**Comment by pangpangshui** [comment]
I found that I did't change the target SDK to API 23. When I changed it, the error message(`W GLConsumer: [SurfaceTexture-0-21322-0] bindTextureImage: clearing GL error: 0x502`) disappeared, but the phone got a black screen just like I couldn't call the camera. And the MIUI didn't give me a tips that permiting to call camera, so I guess Android 6.0 permission would be the reason. But have no idea how to fix it. @188080501    -_-

**Comment by 188080501** [comment]
have no idea too, sorry T.T

**Comment by pangpangshui** [comment]
Okay, thanks anyway. 
By the way, after google it, I think I may be connected with opengl and it is neccesarry to override Java function. I am not familiar with Java, so there's nothing I can do about the problem. May be like this http://maninara.blogspot.sg/2015/03/render-camera-preview-using-opengl-es.html. But I don't know how to modify this project.
Hope someone can fix this problem.

**Comment by pangpangshui** [comment]
Can I ask what version is qzxing that you use? If you use a new one, may be you can fix this problem.

**Comment by 188080501** [comment]
Currently based on zxing development, not qzxing.

**Comment by pangpangshui** [comment]
Thanks, I get it.
If you update your code which can fit Android 6.0, please sent me a message.
Best wishes for you.

**Comment by 188080501** [comment]
you tried to declare the camera permissions?

**Comment by pangpangshui** [comment]
Yes, but I couldn't do it successfully with Java because I am not familiar with Java at all. I have no idea that whether it can work well or not  if I add the camera permission. Hope you can try.

**Comment by pangpangshui** [comment]
I fixed it when I use Qt5.9.1 compiling my project. Thanks.

**Comment by 188080501** [comment]
good news 😄
